pybamm/models/full_battery_models/lithium_ion/base_lithium_ion_model.py

```diff
@@ -285,31 +316,48 @@ def set_crack_submodel(self):
             if crack == "none":
                 self.submodels[
                     domain.lower() + " particle mechanics"
-                ] = pybamm.particle_mechanics.NoMechanics(self.param, domain)
+                ] = pybamm.particle_mechanics.NoMechanics(
+                    self.param, domain, options=self.options, phase="primary"
+                )
             elif crack == "swelling only":
                 self.submodels[
                     domain.lower() + " particle mechanics"
-                ] = pybamm.particle_mechanics.SwellingOnly(self.param, domain)
+                ] = pybamm.particle_mechanics.SwellingOnly(
+                    self.param, domain, options=self.options, phase="primary"
+                )
             elif crack == "swelling and cracking":
                 self.submodels[
                     domain.lower() + " particle mechanics"
                 ] = pybamm.particle_mechanics.CrackPropagation(
-                    self.param, domain, self.x_average
+                    self.param,
+                    domain,
+                    self.x_average,
+                    options=self.options,
+                    phase="primary",
                 )
 
     def set_active_material_submodel(self):
-        for domain in ["Negative", "Positive"]:
-            lam = getattr(self.options, domain.lower())["loss of active material"]
-            if lam == "none":
-                self.submodels[
-                    domain.lower() + " active material"
-                ] = pybamm.active_material.Constant(self.param, domain, self.options)
-            else:
+        for domain in ["negative", "positive"]:
+            lam = getattr(self.options, domain)["loss of active material"]
+            phases = self.options.phase_number_to_names(
+                getattr(self.options, domain)["particle phases"]
+            )
+            for phase in phases:
+                if lam == "none":
+                    submod = pybamm.active_material.Constant(
+                        self.param, domain, self.options, phase
+                    )
+                else:
```

Comment:
They are not, but we still need to loop over all the phases for the constant case since that is where the active material volume fraction and surface area to volume ratio is set. I could reformat this to make it more obvious? But the Imperial guys are adding multi-phase LAM anyway